### DISCUSSION TITLE
How to print out the time taken for each linear iteration

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#27496] [Q&A General]
I need MOOSE to print out the time taken for each linear iteration, but I found that MOOSE by default only prints out the time when the calculation takes a long time. How can I set this up please?

**Comment by GiudGiud** [comment]
Hello

You can reduce the delay to printing the time cost by setting perf graph parameters in the Outputs block

**Comment by hityyds** [comment]
There are just too many parameters to set there. I would quite appreciate it if you can give me more specific advice.

**Reply by GiudGiud** [reply]
in the outputs block, this parameter `perf_graph_live_time_limit` could be set to 1 to get every section to print after 1s

do you need the output live? on the log? (parameters here: https://mooseframework.inl.gov/source/actions/CommonOutputAction.html)
or in the table at the end? You can customize this here
https://mooseframework.inl.gov/source/outputs/PerfGraphOutput.html
